lib/pretty_text.rb

```diff
@@ -340,10 +340,14 @@ def self.extract_links(html)
   def self.excerpt(html, max_length, options = {})
     # TODO: properly fix this HACK in ExcerptParser without introducing XSS
     doc = Nokogiri::HTML.fragment(html)
+    DiscourseEvent.trigger(:before_excerpt, doc, options)
     strip_image_wrapping(doc)
     html = doc.to_html
 
-    ExcerptParser.get_excerpt(html, max_length, options)
+    excerpt = ExcerptParser.get_excerpt(html, max_length, options)
+    DiscourseEvent.trigger(:after_excerpt, excerpt, options)
```

Comment:
I don't think we need 2 new events to handle this case.

I would rename `before_excerpt` to `reduce_excerpt` to be consistent with `reduce_cooked`.

And in the `reduce_excerpt` event, in the poll plugin, I would just replace all occurences of a poll with a link to the post.

Then the `ExcerptParser` will take care of doing the "cutting" where it see fits.